### PR TITLE
Fix ingress metadata.name bug.

### LINKED ISSUE #2920: [stable/wordpress] Ingress bad metadata name with Google Cloud
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Helm and Kubernetes**:
helm: v2.7.0
kube: v1.8.4-gke.0

**Which chart**:
stable/wordpress (0.7.5)

**What happened**:
`googleapi: Error 400: Invalid value for field 'resource.name': 'k8s-um-blog-wp001-blog.example.com-wp001--d8aaf002fa3d533e'. Must be a match of regex '(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)', invalid`

**How to reproduce it** (as minimally and precisely as possible):
Enable ingress, and use Google Cloud.

**Anything else we need to know**:
It could be good to manage a dedicated value for the ingress name if enabled.

### PR DESCRIPTION
**Pull request by ismailbaskin**
It simply fixes #2920 issue and corrects allowEmptyPassword value usage.

### CODE REVIEW

File: stable/wordpress/templates/deployment.yaml [hunk 1]
```diff
@@ -26,11 +26,7 @@ spec:
         imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
         env:
         - name: ALLOW_EMPTY_PASSWORD
-        {{- if .Values.allowEmptyPassword }}
-          value: "yes"
-        {{- else }}
-          value: "no"
-        {{- end }}
+          value: {{ .Values.allowEmptyPassword | quote }}
```
Comment (juan131) [inline stable/wordpress/templates/deployment.yaml:29]: I'm okey with this change but I think it's already quoted on `values.yaml`. I would unquote it on `values.yaml`.

File: stable/wordpress/templates/ingress.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 apiVersion: extensions/v1beta1
 kind: Ingress
 metadata:
-  name: "{{- printf "%s-%s" .name $.Release.Name | trunc 63 | trimSuffix "-" -}}"
+  name: "{{- printf "%s-%s" .name $.Release.Name | trunc 63 | replace "." "-" | trimSuffix "-" -}}"
```
Comment (juan131) [inline stable/wordpress/templates/ingress.yaml:6]: Nice!

**Review by juan131 — COMMENTED** [review]
LGTM! Please take a look to my comment about quoting twice.

File: stable/wordpress/templates/deployment.yaml (hunk 1 above)
Comment (juan131) [reply stable/wordpress/templates/deployment.yaml:29]: You did the opposite. I'd rather leave it as:

```
value: {{ .Values.allowEmptyPassword | quote }}
```

And correct `values.yaml` so it looks like:

```
...
allowEmptyPassword: yes
...
```

File: stable/wordpress/templates/deployment.yaml [hunk 3]
```diff
@@ -26,11 +26,7 @@ spec:
         imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
         env:
         - name: ALLOW_EMPTY_PASSWORD
-        {{- if .Values.allowEmptyPassword }}
-          value: "yes"
-        {{- else }}
-          value: "no"
-        {{- end }}
+          value: {{ .Values.allowEmptyPassword }}
```
Comment (juan131) [inline stable/wordpress/templates/deployment.yaml]: You did the opposite. I'd rather leave it as:

```
value: {{ .Values.allowEmptyPassword | quote }}
```

And correct values.yaml so it looks like:

```
...
allowEmptyPassword: yes
...
```

**Review by juan131 — CHANGES_REQUESTED** [review]

**Review by juan131 — APPROVED** [review]
LGTM! Thanks

File: stable/wordpress/templates/deployment.yaml (hunk 1 above)
Comment (sameersbn) [reply stable/wordpress/templates/deployment.yaml:29]: Having `allowedEmptyPassword: yes` (without quotes) in the `values.yaml` will treat the `allowedEmptyPassword` as a boolean and result in the value `true` for the `ALLOW_EMPTY_PASSWORD` env variable.  The `bitnami/wordpress` image only accepts the value `yes/no` and hence the container launch will fail.

This if block also ensures that `ALLOW_EMPTY_PASSWORD` is assigned the correct value when users specify `--set allowedEmptyPassword=yes` on the helm command line.

**Review by sameersbn — CHANGES_REQUESTED** [review]

File: stable/wordpress/templates/deployment.yaml (hunk 1 above)
Comment (sameersbn) [reply stable/wordpress/templates/deployment.yaml:29]: note: this is also the reason why the tests are failing in the CI (https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/charts/5550/pull-charts-e2e/9800/)

**Pull request closed without merging**